File: Semester4/Advanced Programming/ThriftProj/C#Thrift/Lab2_Festival/Common/domain/entities/Location.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2_Festival.domain.entities
{
    public class Location
    {
        /// <summary>
        /// Constructor for Location class
        /// </summary>
        /// <param name="id">location id</param>
        /// <param name="locName">location name</param>
        /// <param name="totalSeats">number of seats for the given locatio</param>
        public Location(int id, string locName, int totalSeats)
        {
            this.Id = id;
            this.LocName = locName;
            this.TotalSeats = totalSeats;
        }
        ///
        /// Getters and setters
        ///
        public int Id { get; set; }
        public string LocName { get; set; }
        public int TotalSeats { get; set; }


        /// <summary>
        /// Overriding equals method
        /// </summary>
        /// <param name="obj">object to be compared</param>
        /// <returns>true if the objects are equal, false otherwise</returns>
        public override bool Equals(object obj)
        {
            if (!(obj is Location)) return false;

            Location loc = (Location)obj;
            return loc.Id == Id;
        }

        /// <summary>
        /// Overriding hashCode function
        /// </summary>
        /// <returns>Integer value representing the hashCode for the current object</returns>
        public override int GetHashCode()
        {
            var hashCode = 1917924855;
            hashCode = hashCode * -1521134295 + Id.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(LocName);
            hashCode = hashCode * -1521134295 + TotalSeats.GetHashCode();
            return hashCode;
        }
    }
}

File: Semester4/Advanced Programming/Lab2_Hw/src/main/java/domain/entities/Location.java
package domain.entities;

import domain.IHasId;

public class Location implements IHasId<Integer> {
    private Integer locationId;
    private String locName;
    private Integer totalSeats;


    /**
     *
     * @param locId - id of the location
     * @param locName - name of the location
     * @param totalSeats - number of seats in location
     */
    public Location(Integer locId, String locName, Integer totalSeats){
       this.locationId = locId;
       this.locName = locName;
       this.totalSeats = totalSeats;
    }


    /**
     * Getters and setters
     */
    @Override
    public Integer getId() {
        return locationId;
    }

    @Override
    public void setId(Integer integer) {
        this.locationId = integer;
    }

    public Integer getTotalSeats() {
        return totalSeats;
    }

    public String getLocName() {
        return locName;
    }

    public void setLocName(String locName) {
        this.locName = locName;
    }

    public void setTotalSeats(Integer totalSeats) {
        this.totalSeats = totalSeats;
    }

    /**
     * Function to check if two locations are equal
     * @param ot - object to be compared with
     * @return - true fi the objects are equal, false otherwise
     */
    @Override
    public boolean equals(Object ot){
        if(ot == null){
            return false;
        }

        if(this == ot){
            return true;
        }

        if(this.getClass() != ot.getClass()){
            return false;
        }

        Location loc = (Location)ot;

        return loc.getId().equals(this.getId()) &&
               loc.getLocName().equals(this.getLocName()) &&
               loc.getTotalSeats().equals(this.getTotalSeats());
    }
}
